nifi-nar-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/processors/slack/PublishSlack.java

```diff
@@ -0,0 +1,516 @@
+public class PublishSlack extends AbstractProcessor {
+
+    private static final String UPLOAD_ONLY = "upload-only";
+    private static final String MESSAGE_ONLY = "message-only";
+    private static final String CONTENT_MESSAGE = "content-message";
+    private static final String UPLOAD_ATTACH = "upload-attach";
+    private static final String UPLOAD_LINK = "upload-link";
+
+    public static final PropertyDescriptor BOT_TOKEN = new PropertyDescriptor.Builder()
+            .name("Bot Token")
+            .description("The Bot Token that is registered to your Slack application")
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .required(true)
+            .sensitive(true)
+            .build();
+
+    public static final AllowableValue STRATEGY_UPLOAD_ONLY = new AllowableValue(
+            UPLOAD_ONLY,
+            "Upload File Content Only",
+            "Only upload the FlowFile content - it will be a visible Slack message if 'Upload Channel' is specified."
+    );
+
+    public static final AllowableValue STRATEGY_MESSAGE_ONLY = new AllowableValue(
+            MESSAGE_ONLY,
+            "Post Message Only",
+            "Post a simple Slack message with optional 'Text' and/or dynamic Properties as attachments."
+    );
+
+    public static final AllowableValue STRATEGY_CONTENT_MESSAGE = new AllowableValue(
+            CONTENT_MESSAGE,
+            "Post File Content as Message Text",
+            "Post the FlowFile content as a Slack message with optional dynamic Properties as attachments."
+    );
+
+    public static final AllowableValue STRATEGY_UPLOAD_ATTACH = new AllowableValue(
+            UPLOAD_ATTACH,
+            "Upload File Content as Attachment to Message",
+            "Upload the FlowFile content and add a link to the uploaded file as an attachment to the posted Slack message. The message will include optional dynamic Properties as attachments."
+    );
+
+    public static final AllowableValue STRATEGY_UPLOAD_LINK = new AllowableValue(
+            UPLOAD_LINK,
+            "Upload File Content as Embedded Link",
+            "Upload the FlowFile content and append a link to the uploaded file in the posted Slack message. The message will include optional dynamic Properties as attachments."
+    );
+
+    public static final PropertyDescriptor PUBLISH_STRATEGY = new PropertyDescriptor.Builder()
+            .name("publish-strategy")
+            .displayName("Publish Strategy")
+            .description("Determines how the FlowFile is sent to Slack and how it appears.")
+            .allowableValues(STRATEGY_UPLOAD_ONLY, STRATEGY_MESSAGE_ONLY, STRATEGY_CONTENT_MESSAGE, STRATEGY_UPLOAD_ATTACH, STRATEGY_UPLOAD_LINK)
+            .required(true)
+            .defaultValue(UPLOAD_ATTACH)
+            .build();
+
+    public static final PropertyDescriptor CHANNEL = new PropertyDescriptor.Builder()
+            .name("channel")
+            .displayName("Channel")
```

Comment:
Same comment on all of these, using just `.name()` instead of using both a name and a display name.